catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/ProjectUploadRatingDialogTest.java

```diff
@@ -0,0 +1,216 @@
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		bundle = new Bundle();
+		Project project = new Project(context, PROJECT_NAME, false);
+
+		Sprite firstSprite = new SingleSprite("firstSprite");
+		project.getDefaultScene().addSprite(firstSprite);
+		ProjectSaveTask.task(project, context);
+
+		Intent intent = new Intent();
+		intent.putExtra(PROJECT_DIR, project.getDirectory());
+
+		activityTestRule.launchActivity(intent);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		sharedPreferences.edit()
+				.remove(NUMBER_OF_UPLOADED_PROJECTS)
+				.commit();
+	}
+
+	@Test
+	public void testUploadControllerGetsCalled() {
+		sharedPreferences.edit()
+				.putInt(NUMBER_OF_UPLOADED_PROJECTS, 1)
+				.commit();
+
+		onView(withId(R.id.next))
+				.perform(click());
+
+		onView(withText(R.string.next))
+				.perform(click());
+
+		projectUploadController = activityTestRule.getActivity().getProjectUploadController();
+
+		verify(projectUploadController, times(1)).startUpload(eq("ProjectUploadRatingDialogTest"), eq(""), any());
+	}
+
+	@Test
+	public void uploadSuccessRatingDialogShowing() {
+		sharedPreferences.edit()
+				.putInt(NUMBER_OF_UPLOADED_PROJECTS, 1)
+				.commit();
+
+		onView(withId(R.id.next))
+				.perform(click());
+
+		onView(withText(R.string.next))
+				.perform(click());
+
+		InstrumentationRegistry.getInstrumentation().runOnMainSync(() -> {
+			activityTestRule.getActivity().showUploadDialog();
+			activityTestRule.getActivity().onReceiveResult(UPLOAD_RESULT_RECEIVER_RESULT_CODE, bundle);
+		});
+
+		onView(withText(R.string.rating_dialog_rate_now))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void firstUploadSuccessRatingDialogNotShowing() {
+		sharedPreferences.edit()
+				.putInt(NUMBER_OF_UPLOADED_PROJECTS, 0)
+				.commit();
+
+		onView(withId(R.id.next))
+				.perform(click());
+
+		onView(withText(R.string.next))
+				.perform(click());
+
+		InstrumentationRegistry.getInstrumentation().runOnMainSync(() -> {
+			activityTestRule.getActivity().showUploadDialog();
+			activityTestRule.getActivity().onReceiveResult(UPLOAD_RESULT_RECEIVER_RESULT_CODE, bundle);
+		});
+
+		onView(withText(R.string.rating_dialog_rate_now))
+				.check(doesNotExist());
+	}
+
+	@Test
+	public void thirdUploadSuccessRatingDialogNotShowing() {
+		sharedPreferences.edit()
+				.putInt(NUMBER_OF_UPLOADED_PROJECTS, 2)
+				.commit();
+
+		onView(withId(R.id.next))
+				.perform(click());
+
+		onView(withText(R.string.next))
+				.perform(click());
+
+		InstrumentationRegistry.getInstrumentation().runOnMainSync(() -> {
+			activityTestRule.getActivity().showUploadDialog();
+			activityTestRule.getActivity().onReceiveResult(UPLOAD_RESULT_RECEIVER_RESULT_CODE, bundle);
+		});
+
+		onView(withText(R.string.rating_dialog_rate_now))
+				.check(doesNotExist());
+	}
+
+	@Test
+	public void uploadFailRatingDialogNotShowing() {
+		onView(withId(R.id.next))
```

Comment:
I suspect you have to set the number of uploaded projects to `1`, otherwise the rating dialog would not be shown in any case

```
		sharedPreferences.edit()
				.putInt(NUMBER_OF_UPLOADED_PROJECTS, 1)
				.commit();
```